src/hooks/useRegister.jsx

```diff
@@ -0,0 +1,28 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useNavigate } from 'react-router-dom';
+import api, { setTokens } from '../util/auth'; 
+
+export function useRegister() {
```

Comment:
this hooks are used only in SignUpForm component. It's better to create folder SignUpForm, and store all related code inside.